### PR TITLE
Implement curriculum update endpoint

### PR DESCRIPTION
**Pull request by fenix23707**

### CODE REVIEW

**Review by tinmn9088 — APPROVED** [review]

File: academic-server/academic-server-rest/src/main/java/by/kovzov/uis/academic/rest/controller/CurriculumController.java [hunk 1]
```diff
@@ -28,9 +31,24 @@ public Page<CurriculumDto> getAll(@PageableDefault(sort = "approvalDate", direct
...
+    }
+
     @PostMapping
     @PreAuthorize("hasAuthority('CURRICULUM_CREATE')")
     public CurriculumDto create(@RequestBody @Valid CurriculumDto curriculumDto) {
         return curriculumService.create(curriculumDto);
     }
+
+    @PutMapping("/{id}")
+    @PreAuthorize("hasAuthority('CURRICULUM_UPDATEE')")
```
Comment (tinmn9088) [inline academic-server/academic-server-rest/src/main/java/by/kovzov/uis/academic/rest/controller/CurriculumController.java:48]: CURRICULUM_UPDATE

**Review by tinmn9088 — APPROVED** [review]

File: academic-server/academic-server-rest/src/main/java/by/kovzov/uis/academic/rest/controller/CurriculumController.java [hunk 2]
```diff
@@ -28,9 +31,24 @@ public Page<CurriculumDto> getAll(@PageableDefault(sort = "approvalDate", direct
         return curriculumService.getAll(pageable);
     }
 
+    @GetMapping
+    @PreAuthorize("hasAuthority('CURRICULUM_GET')")
+    public Page<CurriculumDto> search(@RequestBody SearchDto searchDto,
```
Comment (tinmn9088) [inline academic-server/academic-server-rest/src/main/java/by/kovzov/uis/academic/rest/controller/CurriculumController.java:36]: If I remember correctly `@RequestBody` doesn't work with `@GetMapping`.

Comment (fenix23707) [reply academic-server/academic-server-rest/src/main/java/by/kovzov/uis/academic/rest/controller/CurriculumController.java:36]: I am not sure either but I don't want to think about this now

File: academic-server/academic-server-rest/src/main/java/by/kovzov/uis/academic/rest/controller/CurriculumController.java (hunk 1 above)
Comment (fenix23707) [reply academic-server/academic-server-rest/src/main/java/by/kovzov/uis/academic/rest/controller/CurriculumController.java]: fixed

**Pull request merged**